assets/src/components/detours/detourModal.tsx

```diff
@@ -0,0 +1,28 @@
+import React from "react"
+import { DiversionPage } from "./diversionPage"
+import { StartDetourProps } from "./detourDropdown"
+
+export const DetourModal = ({
+  detourInfo,
+  onClose,
+  show,
+}: {
+  detourInfo: StartDetourProps
+  onClose: () => void
+  show: boolean
+}) => {
+  return (
```

Comment:
There is a bug that I introduced here, and I'm trying to think about the best way to resolve it.

---

### Steps to Reproduce
 - Open the search maps page.
 - Find a bus, right-click, open the detour modal, and drop a start point.
 - Close the detour modal.
 - Without refreshing, change the zoom level and/or pan the map.
 - Re-open the detour modal from the bus.

### Expected Behavior
 - The start point is gone
 - The map opens to zoom level and center that the search map screen currently shows

### Actual Behavior
 - The start point is still there
 - The map opens to the same zoom level and center that the previous detour modal was previously showing

---

The offending commit is this one, https://github.com/mbta/skate/pull/2377/commits/2dacf205c273e4cd15ef72a30cbdfbb3abe01a6e, which adds a CSS transition to make the modal animate in and out, rather than just _appearing_ (which is pretty jarring, and sort of makes it look like clicking on the dropdown menu didn't do anything).

I'm pretty sure that the root cause is that before ☝️ commit, the modal was being added or removed from the DOM, but afterwards, it's just being given an opacity of 0 and hidden behind everything. That is... in the before-times, React saw the detour modal leave the DOM, and then when it was time to create a new one, it was just a completely brand new one, but in the after-times, as far as React is concerned, it's the same object, so it keeps all of the same state.

I'm thinking that the answer is a `key=` on one of these objects - probably either `DetourModal` or `DiversionPage`. Maybe `key`ing on the timestamp that the user clicked on the dropdown? Or even a random nonce generated when the user clicks the dropdown? I think either of these would work, since they would persist across a user's journey through a single detour modal, and then be different if they right-click again.

Interested to hear what @mbta/skate-developers think about this.

Reply:
`key` is probably the easiest method, and I don't think you'd need to make it very complex, 
```
const [modalKey, setModalKey] = useState(0)

"onModalOpen": () => setModalKey(current => current + 1)
```

Reply:
That makes a lot of sense - giving it a try now...

Reply:
[It worked!](https://github.com/mbta/skate/pull/2377/commits/4b3d067a7b5eaabca2ca6e9d716cf62da23c4d36)